File: src/main/java/io/github/komdosh/alarms/actions/AlarmCustomMin.java
package io.github.komdosh.alarms.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import io.github.komdosh.alarms.services.AlarmService;

import java.util.concurrent.TimeUnit;

public class AlarmCustomMin extends AnAction {

    public AlarmCustomMin() {
        super("Alarm20Min");
    }

    public void actionPerformed(AnActionEvent event) {
        Project p = event.getProject();
        String s = Messages.showInputDialog(p, "Set scheduler minutes delay for alarm:",
            "Alarm Is Set", Messages.getInformationIcon());

        try {
            if (s == null) {
                throw new NumberFormatException("String is null");
            }
            int minutes = Integer.parseInt(s);

            AlarmService alarmService = ServiceManager.getService(AlarmService.class);
            alarmService.setAlertInSeconds(p, TimeUnit.SECONDS.toSeconds(minutes), "Minutes");
        } catch (NumberFormatException e) {
            Messages.showMessageDialog(p, "Wrong number of minutes",
                "Alarm Is Not Set", Messages.getErrorIcon());
        }

    }
}

File: src/main/kotlin/io/github/komdosh/alarms/actions/AlarmCustomMin.kt
package io.github.komdosh.alarms.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.ui.Messages
import io.github.komdosh.alarms.services.AlarmService
import java.util.concurrent.TimeUnit

class AlarmCustomMin : AnAction("AlarmCustomMin") {
    override fun actionPerformed(event: AnActionEvent) {
        event.project?.let {
            val minutesString = Messages.showInputDialog(it, "Set scheduler minutes delay for alarm",
                    "Alarm Set Custom Delay", Messages.getInformationIcon())

            try {
                if (minutesString.isNullOrBlank()) {
                    throw NumberFormatException("String couldn't be blank or null")
                }
                val minutes = Integer.parseInt(minutesString)

                val alarmService = ServiceManager.getService(AlarmService::class.java)
                alarmService.setAlertInSeconds(it, TimeUnit.MINUTES, minutes.toLong())
            } catch (e: NumberFormatException) {
                Messages.showMessageDialog(it, "Wrong number of minutes",
                        "Alarm Is Not Set", Messages.getErrorIcon())
            }
        }
    }
}
